DecipherNow/gm-fabric-dashboard
# Improve Dev Workflow

Resolves #1310

## Proposed Changes

This PR uses npm-run-all and Husky to simplify the way that the dev environment works. Running npm start now starts all UI and backend mocks locally and configures the meta tag needed to tell the UI how to reach the backend. When the dev hits CTRL+C to send a SIGINT, the poststart npm lifecycle hook performs cleanup activities to put the repo back in a pristine state. When the dev performs a commit, Husky controlled git hooks ensures that the index.html meta tags are not set and, if an .isRunning file is found in the root directory, sets the metatag back to localhost so that the dev environment continues to work as expected.

## How to test
1. Ensure that the meta tag is not set in `index.html`
2. Ensure that the hidden file `.isRunning` is not present in the project root
3. Run `npm start` and watch the dev console for errors
4. Ensure that the meta tag has been set
5. Ensure that the hidden file `.isRunning` was created in the project root
6. Try to commit the modified `index.html`
7. View the file changes since the last commit by running `git diff HEAD^ HEAD` and confirm that the result is empty
8. Hit Ctrl+C to exit dev mode
9. Ensure that the meta tag is not set in `index.html`
10. Ensure that the hidden file `.isRunning` is not present in the project root
